driver-mapping/src/test/java/com/datastax/driver/mapping/config/MappingConfigurationPropertyAccessTest.java

```diff
@@ -0,0 +1,211 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.mapping.config;
+
+import com.datastax.driver.core.CCMTestsSupport;
+import com.datastax.driver.mapping.DefaultMappedProperty;
+import com.datastax.driver.mapping.MappedProperty;
+import com.datastax.driver.mapping.Mapper;
+import com.datastax.driver.mapping.MappingManager;
+import com.datastax.driver.mapping.annotations.PartitionKey;
+import com.datastax.driver.mapping.annotations.Table;
+import com.datastax.driver.mapping.annotations.Transient;
+import com.google.common.collect.Sets;
+import com.google.common.reflect.TypeToken;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Test for JAVA-1310 - validate ability configure property scope - getters vs. fields
+ */
+public class MappingConfigurationPropertyAccessTest extends CCMTestsSupport {
+
+    @Override
+    public void onTestContextInitialized() {
+        execute("CREATE TABLE foo (k int primary key, v int)");
+        execute("INSERT INTO foo (k, v) VALUES (1, 1)");
+    }
+
+    @Test(groups = "short")
+    public void should_ignore_fields() {
+        MappingConfiguration conf = MappingConfiguration.builder()
+                .withPropertyAccessStrategy(new GetterSetterOnlyPropertyAccessStrategy())
+                .build();
+        MappingManager mappingManager = new MappingManager(session(), conf);
+        mappingManager.mapper(Foo1.class);
+    }
+
+    @Table(name = "foo")
+    @SuppressWarnings({"unused", "WeakerAccess"})
+    public static class Foo1 {
+        private int k;
+
+        private int notAColumn;
+
+        @PartitionKey
+        public int getK() {
+            return k;
+        }
+
+        public void setK(int k) {
+            this.k = k;
+        }
+    }
+
+    @Test(groups = "short")
+    public void should_ignore_getters() {
+        MappingConfiguration conf = MappingConfiguration.builder()
+                .withPropertyAccessStrategy(new FieldOnlyPropertyAccessStrategy())
```

Comment:
Seeing this in use, it looks a lot less complicated.  User only has to pass in the strategy they are interested in, and the provided implementations seem pretty viable and should cover a majority of use cases I would hope 👍